cmd/img_mgmt/port/zephyr/src/zephyr_img_mgmt.c

```diff
@@ -412,6 +428,24 @@ int img_mgmt_impl_erase_if_needed(uint32_t off, uint32_t len)
 }
 #endif
 
+int
+img_mgmt_impl_swap_type_multi(int slot)
+{
+    switch (mcuboot_swap_type_multi(slot_to_image_index(slot))) {
```

Comment:
The mcuboot_swap_type_multi depends on https://github.com/nrfconnect/sdk-zephyr/pull/593, specifically on https://github.com/nrfconnect/sdk-zephyr/pull/593/commits/baa91b39c33e67489cb73f98320748dd07f72a26 which is "noup"; this will not work with zephyr, only nrf.

Reply:
I was unsure if https://github.com/nrfconnect/sdk-zephyr/commit/baa91b39c33e67489cb73f98320748dd07f72a26#diff-696b0049cfb981ee66f0191ccaf755f25a632c35b86da7dc1d739c3de8a60ed3R147 would be accepted upstream as it limits the functionality to two images. 

Since I'm on vacation could you take this further?

Reply:
Yes I will.